test/sys/test_signalfd.rs

```diff
@@ -0,0 +1,52 @@
+#[cfg(target_os = "linux")]
+#[test]
+// FIXME: Fails with "fatal runtime error: failed to initiate panic, error 5" on i686-unknown-linux-musl
+#[cfg_attr(all(target_env = "musl", target_arch = "x86"), ignore)]
+fn test_signalfd() {
+    use nix::sys::signalfd::{SFD_NONBLOCK, SignalFd};
+    use nix::sys::signal;
+    use nix::unistd;
+    use nix::unistd::*;
+    use nix::unistd::ForkResult::*;
+    use nix::sys::signal::*;
+    use nix::sys::wait::*;
+
+    // Set up a signalfd to listen for SIGUSR1 on this thread
+    let mut mask = SigSet::empty();
+    mask.add(SIGUSR1);
```

Comment:
I was going to use `SIGUSR2`, but I actually think this implementation isn't quite right. According to [this](https://stackoverflow.com/questions/11679568/signal-handling-with-multiple-threads-in-linux#11679770] signal disposition is shared across all threads in a process. So I think we should actually have a `SIGNAL_MTX` and not one specific to each signal.